src/components/tables/TableHeader.tsx

```diff
@@ -1,17 +1,27 @@
 import { ITableHeaderCellProps, TableHeaderCell } from './TableHeaderCell';
+import { TableHeaderCellConnected } from './TableHeaderCellConnected';
 import * as React from 'react';
 import * as _ from 'underscore';
 
 export interface ITableHeaderProps extends React.ClassAttributes<TableHeader> {
   columns: ITableHeaderCellProps[];
   headerClass?: string;
+  connectCell?: true;
```

Comment:
I've created `IReduxStatePossibleProps` from which you can extend instead. It's a simple interface:
```
export interface IReduxStatePossibleProps {
  withReduxState?: boolean;
}
```
It's exactly the same, but it's might be better to use the same thing everywhere. Also I don't think there is a `TableHeaderConnected`, so this won't apply here, but usually I let the connected component set this value to true or false.

Reply:
i'm not sure of what you want me to do here, I have an idea, but I think the current implementation is good enough. Like you said, `withReduxState` is not applied on the TableHeader here, and if I put it at the TableHeaderCell level, I would have to add a kind of TableHeaderCellFactory component to add the `withReduxState` logic in it, and return the connected TableHeaderCell or unconnected one depending on the props passed to it.

Reply:
`ITableHeaderProps` should extend `IReduxStatePossibleProps` so that all components that can be toggled from not connected to connected all work the same way.

Reply:
changed, let me know what you think